src/ActiveQuery.php

```diff
@@ -152,15 +152,11 @@ public function snippetByModel()
      * Executes query and returns all results as an array.
      * @param Connection $db the DB connection used to create the DB command.
      * If null, the DB connection returned by [[modelClass]] will be used.
-     * @return array the query results. If the query results in nothing, an empty array will be returned.
+     * @return array|ActiveRecord[] the query results. If the query results in nothing, an empty array will be returned.
      */
     public function all($db = null)
     {
-        if ($this->emulateExecution) {
-            return [];
-        }
-        $rows = $this->createCommand($db)->queryAll();
-        return $this->populate($rows);
+        return parent::all($db);
```

Comment:
If there's nothing else than calling parent method happens, does it make sense to remove the method?

Reply:
Purpose of this override - change method's return type, add `ActiveRecord[]` to parent's return type. IMHO.

Reply:
OK.